internal/pkg/composable/controller.go

```diff
@@ -35,18 +35,24 @@ type Controller interface {
 	// Watch returns the channel to watch for variable changes.
 	Watch() <-chan []*transpiler.Vars
 
-	// Close closes the controller, allowing for any resource
-	// cleanup and such.
-	Close()
+	// Observe instructs the variables to observe.
+	Observe([]string)
 }
 
 // controller manages the state of the providers current context.
 type controller struct {
-	logger           *logger.Logger
-	ch               chan []*transpiler.Vars
-	errCh            chan error
-	contextProviders map[string]*contextProviderState
-	dynamicProviders map[string]*dynamicProviderState
+	logger          *logger.Logger
+	ch              chan []*transpiler.Vars
+	observedCh      chan map[string]bool
+	errCh           chan error
+	restartInterval time.Duration
+
+	managed          bool
+	contextProviders map[string]contextProvider
+	dynamicProviders map[string]dynamicProvider
```

Comment:
These are more like provider *factories* rather than providers themselves, right?

Reply:
Correct.

Reply:
Updated variable naming to make it easier to read.